enumerable.rb

```diff
@@ -0,0 +1,118 @@
+module Enumerable
+  def my_each
+    return enum_for(:my_each) unless block_given?
+
+    i = 0
+    while i < to_a.length
+      yield(to_a[i])
+      i += 1
+    end
+    self
+  end
+
+  def my_each_with_index
+    return enum_for(:my_each_with_index) unless block_given?
+
+    length.times do |x|
+      yield(self[x], x)
+    end
+  end
+
+  def my_select
+    return enum_for(:my_select) unless block_given?
+
+    arr = []
+    to_a.length.times do |x|
+      arr.push(self[x]) if yield(self[x])
+    end
+    arr
+  end
```

Comment:
### my_select

- [ ] when self is range raises the following error `undefined method '[]' for range`
#### hint
- make sure you don't index a range using `[]`, as it is not available for range.